tools/c7n_mailer/c7n_mailer/azure_mailer/sendgrid_delivery.py

```diff
@@ -94,18 +96,19 @@ def sendgrid_handler(self, queue_message, to_addrs_to_email_messages_map):
 
         for email_to_addrs, message in six.iteritems(to_addrs_to_email_messages_map):
             for to_address in email_to_addrs:
+                message_string = base64.b64decode(message.get_payload()).decode('utf-8')
```

Comment:
Will follow up offline about this.  We should use sendgrids Mail.from_emailmessage() method and replace most of this code.

Reply:
It wasn't quite as easy as calling `Mail.from_EmailMessage`. So, I copied that function at the bottom of this file and made some changes.

1. I think SendGrid has a bug in that call which is fixed by creating a `To` object instead of an `Email` object when calling the Mail constructor.
1. We set a number of headers on the message object that break the SendGrid API (they are reserved keys), so I ignore those headers when copying the headers over. [more info](https://sendgrid.com/docs/API_Reference/Web_API_v3/Mail/errors.html#message.headers)